packages/run-protocol/test/test-gov-collateral.js

```diff
@@ -0,0 +1,452 @@
+import { test as anyTest } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import process from 'process';
+import url from 'url';
+import path from 'path';
+import { E, Far } from '@endo/far';
+import {
+  addBankAssets,
+  makeAddressNameHubs,
+  makeOracleBrands,
+  makeBoard,
+  startPriceAuthority,
+} from '@agoric/vats/src/core/basic-behaviors.js';
+import {
+  bridgeCoreEval,
+  makeClientManager,
+} from '@agoric/vats/src/core/chain-behaviors.js';
+import { extractCoreProposalBundles } from '@agoric/deploy-script-support/src/extract-proposal.js';
+import { makeCoreProposalBehavior } from '@agoric/deploy-script-support/src/coreProposalBehavior.js';
+import { makeNameHubKit } from '@agoric/vats/src/nameHub.js';
+import { AmountMath, makeIssuerKit } from '@agoric/ertp';
+import { makeNodeBundleCache } from './bundleTool.js';
+import { setupBootstrap, setUpZoeForTest } from './supports.js';
+
+const dirname = url.fileURLToPath(new URL('.', import.meta.url));
+
+/** @type {import('ava').TestInterface<Awaited<ReturnType<makeTestContext>>>} */
+// @ts-expect-error cast
+const test = anyTest;
+
+const contractRoots = {
+  mintHolder: '../vats/src/mintHolder.js',
+  econCommitteeCharter: './src/econCommitteeCharter.js',
+};
+
+const coreProposals = {
+  addCollateral: '../scripts/add-collateral-core.js',
+  startRunPreview: '../scripts/init-core.js',
+  inviteCommittee: '../scripts/invite-committee-core.js',
+};
+
+const voterAddresses = {
+  Rowland: `agoric1qed57ae8k5cqr30u5mmd46jdxfr0juyggxv6ad`,
+  Bill: `agoric1xgw4cknedau6xhrlyn6c8e40d02mejee8gwnef`,
+  Dan: `agoric1yumvyl7f5nkalss7w59gs6n3jtqv5gmarudx55`,
+};
+
+const makeTestContext = async () => {
+  const bundleCache = await makeNodeBundleCache('bundles/', s => import(s));
+  const { zoe, feeMintAccess } = setUpZoeForTest();
+
+  const runIssuer = await E(zoe).getFeeIssuer();
+  const runBrand = await E(runIssuer).getBrand();
+
+  const install = (src, dest) =>
+    bundleCache.load(src, dest).then(b => E(zoe).install(b));
+  const installation = {
+    mintHolder: install(contractRoots.mintHolder, 'mintHolder'),
+    econCommitteeCharter: install(
+      contractRoots.econCommitteeCharter,
+      'econCommitteeCharter',
+    ),
+  };
+
+  const bundlePathToInstallP = new Map();
+  const cachedInstall = (src, dest) => {
+    if (!dest) {
+      dest = src.replace(/(\\|\/|:)/g, '_');
+    }
+    if (!bundlePathToInstallP.has(dest)) {
+      const match = path.basename(dest).match(/^bundle-(.*)\.js$/);
+      const bundle = match ? match[1] : dest;
+      bundlePathToInstallP.set(dest, install(src, bundle));
+    }
+    return bundlePathToInstallP.get(dest);
+  };
+
+  return {
+    cachedInstall,
+    cleanups: [],
+    zoe: await zoe,
+    feeMintAccess: await feeMintAccess,
+    runKit: { brand: runBrand, issuer: runIssuer },
+    installation,
+  };
+};
+
+test.before(async t => {
+  t.context = await makeTestContext();
+});
+
+const makeScenario = async t => {
```

Comment:
each test begins with `const s = await makeScenario(t);` consider in `test.before`,
```
t.context.scenario = await makeScenario(t);
```
then within test,
```
const { s } = t.context.scenario;
```

Reply:
I think I put as much stuff as is shareable in `t.context`. `makeScenario()` makes mutable stuff; if we shared that between tests, it would be bad, I think.

Reply:
Ah right. It would have to be in `test.beforeEach`. https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#test-context